### PR TITLE
Implement stat increases upon each level up in CharacterInfo class

### PR DESCRIPTION
**Pull request by warpfox**
#20

### CODE REVIEW

File: Assets/Scripts/CharacterInfo.cs [hunk 1]
```diff
@@ -36,42 +31,122 @@ public class CharacterInfo : MonoBehaviour {
...
-		vitality += Random.Range(1,6);
-		speed += Random.Range(1,6);
-		offense += Random.Range(1,6);
-		defense += Random.Range(1,6);
-		#endregion
-		
-		if (vitality % 5 == 0){
-			maxHP++;
+		#region Character Specific Stat Growth Rates
+		if (characterName == CharacterName.Ness){
```
Comment (VirtuaBoza) [inline Assets/Scripts/CharacterInfo.cs]: This is the perfect case for a switch block

File: Assets/Scripts/CharacterInfo.cs [hunk 2]
```diff
@@ -36,42 +31,122 @@ public class CharacterInfo : MonoBehaviour {
...
-		luck += Random.Range(1,6);
-		vitality += Random.Range(1,6);
-		speed += Random.Range(1,6);
-		offense += Random.Range(1,6);
-		defense += Random.Range(1,6);
-		#endregion
-		
-		if (vitality % 5 == 0){
-			maxHP++;
+		#region Character Specific Stat Growth Rates
```
Comment (VirtuaBoza) [inline Assets/Scripts/CharacterInfo.cs]: I would say, if you need to region a block of code in a method, it probably needs to be extracted into its own method

Comment (VirtuaBoza) [inline Assets/Scripts/CharacterInfo.cs]: I would say, if you need to region a block of code in a method, it probably needs to be extracted into its own method

File: Assets/Scripts/CharacterInfo.cs (hunk 1 above)
Comment (VirtuaBoza) [inline Assets/Scripts/CharacterInfo.cs]: This is the perfect case for a switch block

Comment (warpfox) [reply Assets/Scripts/CharacterInfo.cs]: I finally get to use it!

File: Assets/Scripts/CharacterInfo.cs (hunk 2 above)
Comment (warpfox) [reply Assets/Scripts/CharacterInfo.cs]: Hm. I'm actually thinking that I could store these "growth rates" in readonly character-specific arrays and then just call the indexes for the actual stat-increasing part. That would, hopefully, refactor it into a new method or two, and would be a little more elegant, I think.

**Pull request merged**